.github/workflows/publish.yaml

```diff
@@ -0,0 +1,45 @@
+# .github/workflows/chromatic.yml
+
+# Workflow name
+name: 'Publish Component Library'
+
+# Event for the workflow
+on:
+  pull_request_target:
+    branches:
+      - master
+    types:
+      - closed
+    paths:
+      - packages/components/**            
+  push:
+    branches-ignore:
+      - 'master'
+    paths:
+      - packages/components/**      
+
+# List of jobs
+jobs:
+  npm-publish:
+    # Operating System
+    runs-on: ubuntu-latest
+    # Job steps
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies, build
+        run: yarn workspace @calyptia-vivo/components install && yarn workspace @calyptia-vivo/components build
+      - name: Configure git
+        run: git config --global user.name "Continuous Integration" && git config --global user.email "ci@calyptia.com"         
+      - name: Publish master
+        if: ${{ github.event_name != 'push' && github.event.pull_request.merged == true }} 
+        run: yarn workspace @calyptia-vivo/components version --patch && yarn workspace @calyptia-vivo/frontend add @calyptia-vivo/components@$npm_package_version
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.VIVO_NPM_AUTH_TOKEN }}                
+      - name: Publish staging    
+        run: yarn workspace @calyptia-vivo/components version --preid pr --prerelease && yarn workspace @calyptia-vivo/frontend add @calyptia-vivo/components@$npm_package_version
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.VIVO_NPM_AUTH_TOKEN }}
+        if: ${{ github.event_name == 'push' && github.event.pull_request.merged == false }}
+      - name: Commit package update
+        run: 
+          git commit -a -m "Components Version Updated && git push
```

Comment:
Pushing straight to `master`?
Normally we do a PR via an action to give a review but I'm not sure if that's what you want: https://github.com/fluent/fluent-bit/blob/c4cfc1babe27a4c4583f0f21e13a6a433a46f6ae/.github/workflows/staging-release.yaml#L868-L881